elastic/integrations
# [azure_network_watcher_vnet] Initial release of the azure network watcher vnet

## Proposed commit message
Create New integration package azure_network_watcher_vnet.
- Added data stream.
- Added data collection logic for log data stream.
- Added the ingest pipeline for log data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for log data stream.
- Added system test cases for log data stream.


## Checklist



## How to test this PR locally



- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/azure_network_watcher_vnet directory.
- Run the following command to run tests. 
> elastic-package test 

```
--- Test results for package: azure_network_watcher_vnet - START ---
╭────────────────────────────┬─────────────┬───────────┬───────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME     │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼───────────────┼────────┼──────────────┤
│ azure_network_watcher_vnet │ log         │ pipeline  │ test-vnet.log │ PASS   │    4.04771ms │
╰────────────────────────────┴─────────────┴───────────┴───────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_vnet - END   ---
Done
--- Test results for package: azure_network_watcher_vnet - START ---
╭────────────────────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ azure_network_watcher_vnet │ log         │ static    │ Verify sample_event.json │ PASS   │ 102.135753ms │
╰────────────────────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_vnet - END   ---
Done
--- Test results for package: azure_network_watcher_vnet - START ---
╭────────────────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ azure_network_watcher_vnet │ log         │ system    │ default   │ PASS   │ 37.890174557s │
╰────────────────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: azure_network_watcher_vnet - END   ---
Done
--- Test results for package: azure_network_watcher_vnet - START ---
╭────────────────────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                                           │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ azure_network_watcher_vnet │             │ asset     │ dashboard azure_network_watcher_vnet-5ecce30b-1eba-4bd8-b440-cd266263449a is loaded │ PASS   │      1.093µs │
│ azure_network_watcher_vnet │             │ asset     │ search azure_network_watcher_vnet-0e5d4cba-9f41-4c45-b858-54d3e83df6e5 is loaded    │ PASS   │        118ns │
│ azure_network_watcher_vnet │             │ asset     │ search azure_network_watcher_vnet-51e7dbb1-2c87-429a-8ffa-f1c78ceb2eab is loaded    │ PASS   │        116ns │
│ azure_network_watcher_vnet │             │ asset     │ search azure_network_watcher_vnet-b52c9df2-cb7f-4ebe-af8b-569c213423cf is loaded    │ PASS   │        125ns │
│ azure_network_watcher_vnet │ log         │ asset     │ index_template logs-azure_network_watcher_vnet.log is loaded                        │ PASS   │        271ns │
│ azure_network_watcher_vnet │ log         │ asset     │ ingest_pipeline logs-azure_network_watcher_vnet.log-0.1.0 is loaded                 │ PASS   │        164ns │
╰────────────────────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: azure_network_watcher_vnet - END   ---
Done
```


## Screenshot
![image](https://github.com/elastic/integrations/assets/123942854/64ad171a-5d09-4e02-9584-1631c4d62d54)
![image](https://github.com/elastic/integrations/assets/123942854/67ed0745-2e83-4268-8158-3e35000484b2)
